Extensions/CMake/SlicerExtensionPackageAndUploadTarget.cmake

```diff
@@ -98,12 +103,6 @@ if(NOT PACKAGEUPLOAD)
     Slicer_REVISION
     EXTENSION_NAME
     EXTENSION_CATEGORY
```

Comment:
For extension built using the extension build-system, we will still need to propagate the extension category and ensure it is serialized in the `s4ext` file bundled in the extension, without doing so, extension installed from file would end up without any category.

Reply:
After further consideration, this is not a problem. Since modules are categorized independently of the extension, the associated modules will be organized whether the extension is installed from files or using the extension manager.

Reply:
`EXTENSION_NAME`, `EXTENSION_CATEGORY` and `EXTENSION_ENABLED` are the one expected to be extracted from the json files and used here, we should not rely on the value already set in the `CMakeLists.txt`. 

I need to update the current approach to effectively ensure it is the case.